pkg/psmdb/mongos.go

```diff
@@ -0,0 +1,361 @@
+				RestartPolicy:     corev1.RestartPolicyAlways,
+				ImagePullSecrets:  cr.Spec.ImagePullSecrets,
+				Containers:        []corev1.Container{c},
+				InitContainers:    initContainers,
+				Volumes:           volumes(cr),
+				SchedulerName:     cr.Spec.SchedulerName,
+			},
+		},
+	}, nil
+}
+
+func initContainers(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod) []corev1.Container {
+	inits := []corev1.Container{}
+	if cr.CompareVersion("1.5.0") >= 0 {
+		inits = append(inits, EntrypointInitContainer(operatorPod.Spec.Containers[0].Image))
+	}
+
+	return inits
+}
+
+func mongosContainer(cr *api.PerconaServerMongoDB) (corev1.Container, error) {
+	fvar := false
+
+	resources, err := CreateResources(cr.Spec.Sharding.Mongos.ResourcesSpec)
+	if err != nil {
+		return corev1.Container{}, fmt.Errorf("resource creation: %v", err)
+	}
+
+	volumes := []corev1.VolumeMount{
+		{
+			Name:      MongodDataVolClaimName,
+			MountPath: MongodContainerDataDir,
+		},
+		{
+			Name:      InternalKey(cr),
+			MountPath: mongodSecretsDir,
+			ReadOnly:  true,
+		},
+		{
+			Name:      "ssl",
+			MountPath: sslDir,
+			ReadOnly:  true,
+		},
+		{
+			Name:      "ssl-internal",
+			MountPath: sslInternalDir,
+			ReadOnly:  true,
+		},
+	}
+
+	if *cr.Spec.Mongod.Security.EnableEncryption {
+		volumes = append(volumes,
+			corev1.VolumeMount{
+				Name:      cr.Spec.Mongod.Security.EncryptionKeySecret,
+				MountPath: mongodRESTencryptDir,
+				ReadOnly:  true,
+			},
+		)
+	}
+
+	container := corev1.Container{
+		Name:            "mongos",
+		Image:           cr.Spec.Image,
+		ImagePullPolicy: cr.Spec.ImagePullPolicy,
+		Args:            mongosContainerArgs(cr, resources),
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          mongosPortName,
+				HostPort:      cr.Spec.Sharding.Mongos.HostPort,
+				ContainerPort: cr.Spec.Sharding.Mongos.Port,
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name:  "SERVICE_NAME",
+				Value: cr.Name,
+			},
+			{
+				Name:  "NAMESPACE",
+				Value: cr.Namespace,
+			},
+			{
+				Name:  "MONGODB_PORT",
+				Value: strconv.Itoa(int(cr.Spec.Sharding.Mongos.Port)),
+			},
+		},
+		EnvFrom: []corev1.EnvFromSource{
+			{
+				SecretRef: &corev1.SecretEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cr.Spec.Secrets.Users,
+					},
+					Optional: &fvar,
+				},
+			},
+		},
+		WorkingDir:      MongodContainerDataDir,
+		LivenessProbe:   &cr.Spec.Sharding.Mongos.LivenessProbe.Probe,
+		ReadinessProbe:  cr.Spec.Sharding.Mongos.ReadinessProbe,
+		SecurityContext: cr.Spec.Sharding.Mongos.ContainerSecurityContext,
+		Resources:       resources,
+		VolumeMounts:    volumes,
+	}
+
+	if cr.CompareVersion("1.5.0") >= 0 {
+		container.EnvFrom = []corev1.EnvFromSource{
+			{
+				SecretRef: &corev1.SecretEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "internal-" + cr.Name + "-users",
+					},
+					Optional: &fvar,
+				},
+			},
+		}
```

Comment:
1) why `if` needed?
2) why env needed?

Reply:
`if` not needed https://github.com/percona/percona-server-mongodb-operator/commit/be7bdefaa2b6f362495a8a77e1b79e5e50ad2560
but users needed for auth